packages/xsnap/src/avaAssertXS.js

```diff
@@ -0,0 +1,351 @@
+/* eslint-disable no-await-in-loop */
+// @ts-check
+
+/** global print */
+
+const { freeze, keys } = Object;
+
+// ack Paul Roub Aug 2014
+// https://stackoverflow.com/a/25456134/7963
+/** @type {(x: unknown, y: unknown) => boolean } */
+function deepEqual(x, y) {
+  if (Object.is(x, y)) {
+    return true;
+  }
+  if (
+    typeof x === 'object' &&
+    x != null &&
+    typeof y === 'object' &&
+    y != null
+  ) {
+    if (keys(x).length !== keys(y).length) {
+      const detail = JSON.stringify({
+        actual: {
+          length: keys(x).length,
+          keys: keys(x),
+        },
+        expected: {
+          length: keys(y).length,
+          keys: keys(y),
+        },
+      });
+      throw new Error(`Object keys length: ${detail}`);
+    }
+
+    const { hasOwnProperty } = Object.prototype;
+    /** @type {(obj: Object, prop: string | symbol | number) => boolean} */
+    const hasOwnPropertyOf = (obj, prop) =>
+      Reflect.apply(hasOwnProperty, obj, [prop]);
+    for (const prop of Reflect.ownKeys(x)) {
+      if (hasOwnPropertyOf(y, prop)) {
+        if (!deepEqual(x[prop], y[prop])) {
+          return false;
+        }
+      } else {
+        // Separately, should this have the same detail structure as the other errors you're throwing?
+        throw new Error(`missing property ${String(prop)}`);
+      }
+    }
+
+    return true;
+  }
+  const detail = JSON.stringify({
+    actual: { type: typeof x, value: x },
+    expected: { type: typeof y, value: y },
+  });
+  throw new Error(detail);
+}
+
+/**
+ * Test status reporting inspired by Test Anything Protocol (TAP)
+ *
+ * ref https://testanything.org/tap-specification.html
+ *
+ * @param {(msg: TapMessage) => void} send
+ *
+ * @typedef { ReturnType<typeof tapFormat> } TapFormat
+ * @typedef {import('./avaXS').TapMessage} TapMessage
+ */
+function tapFormat(send) {
+  return freeze({
+    /** @type {(qty: number) => void} */
+    plan(qty) {
+      send({ plan: qty });
+    },
+    /** @type {(n: number, t?: string) => void} */
+    ok(testNum, txt) {
```

Comment:
Parameter names in type should be the same as the parameter names in the function unless there's a reason why not.
```suggestion
    /** @type {(testNum: number, txt?: string) => void} */
    ok(testNum, txt) {
```
Please fix everywhere this arises without a reason not to.

Reply:
this was me expressing my frustration that tsc doesn't infer the type by itself and that it requires names for no evident reason.

it seemed good enough for test code. but ok...